consensus/polybft/contractsapi/init.go

```diff
@@ -0,0 +1,161 @@
+
+	L1ExitTestABI *abi.ABI
+)
+
+func init() {
+	var err error
+
+	CheckpointManager, err = artifact.DecodeArtifact([]byte(CheckpointManagerArtifact))
+	if err != nil {
+		panic(err)
+	}
+
+	ExitHelper, err = artifact.DecodeArtifact([]byte(ExitHelperArtifact))
+	if err != nil {
+		panic(err)
+	}
+
+	L2StateSender, err = artifact.DecodeArtifact([]byte(L2StateSenderArtifact))
+	if err != nil {
+		panic(err)
+	}
+
+	BLS, err = artifact.DecodeArtifact([]byte(BLSArtifact))
+	if err != nil {
+		panic(err)
+	}
+
+	BLS256, err = artifact.DecodeArtifact([]byte(BN256G2Artifact))
+	if err != nil {
+		panic(err)
+	}
+
+	StateSender, err = artifact.DecodeArtifact([]byte(StateSenderArtifact))
+	if err != nil {
+		panic(err)
+	}
+
+	StateReceiver, err = artifact.DecodeArtifact([]byte(StateReceiverArtifact))
+	if err != nil {
+		panic(err)
+	}
+
+	err = json.Unmarshal([]byte(l1ContractABIBytes), &L1ExitTestABI)
+	if err != nil {
+		panic(err)
+	}
+}
+
+/*
+	contract L1 {
+	    uint256 public id;
+	    address public addr;
+	    bytes public data;
+	    uint256 public counter;
+	    function onL2StateReceive(uint256 _id, address _addr,bytes memory _data) public {
+	        id=_id;
+	        addr=_addr;
+	        data=_data;
+	        counter++;
+	    }
+	}
+*/
+const l1ContractABIBytes = `[
+    {
+      "inputs": [],
+      "name": "addr",
+      "outputs": [
+        {
+          "internalType": "address",
+          "name": "",
+          "type": "address"
+        }
+      ],
+      "stateMutability": "view",
+      "type": "function"
+    },
+    {
+      "inputs": [],
+      "name": "counter",
+      "outputs": [
+        {
+          "internalType": "uint256",
+          "name": "",
+          "type": "uint256"
+        }
+      ],
+      "stateMutability": "view",
+      "type": "function"
+    },
+    {
+      "inputs": [],
+      "name": "data",
+      "outputs": [
+        {
+          "internalType": "bytes",
+          "name": "",
+          "type": "bytes"
+        }
+      ],
+      "stateMutability": "view",
+      "type": "function"
+    },
+    {
+      "inputs": [],
+      "name": "id",
+      "outputs": [
+        {
+          "internalType": "uint256",
+          "name": "",
+          "type": "uint256"
+        }
+      ],
+      "stateMutability": "view",
+      "type": "function"
+    },
+    {
+      "inputs": [
+        {
+          "internalType": "uint256",
+          "name": "_id",
+          "type": "uint256"
+        },
+        {
+          "internalType": "address",
+          "name": "_addr",
+          "type": "address"
+        },
+        {
+          "internalType": "bytes",
+          "name": "_data",
+          "type": "bytes"
+        }
+      ],
+      "name": "onL2StateReceive",
+      "outputs": [],
+      "stateMutability": "nonpayable",
+      "type": "function"
+    }
+  ]
```

Comment:
Does it make sense to put this part to a separate file as well as `L1ExitTestBytecode` and `L1ExitDeployedBytecode`?

Also it would be great if SC team eventually would implement such SC on their end (they did the same thing with `MRC20.sol` basically) and we can use that one for testing purposes.

Reply:
MRC20 - uses native token precompile it uses not only for testing. I'm afraid we'll have to support some testing contracts on our side. I suppose we can use Burn contracts on L1 for that purpose, but it's not implemented yet.
 The main reason why i decided to put it just in a variable - simplicity. I'd say that we need a separate issue about organising compilation of testing contracts.